kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -317,7 +317,8 @@ class KyuubiSessionImpl(
   }
 
   def checkEngineConnectionAlive(): Boolean = {
-    if (Option(client).exists(_.engineConnectionClosed)) return false
+    if (client == null) return true // client has not been initialized
```

Comment:
I'm unclear on the intention of this expression compared to its previous version.

when client is null, `Option(client).exists(_.engineConnectionClosed)` should be false?

Reply:
> when client is null, `Option(client).exists(_.engineConnectionClosed)` should be false?

correct, then it goes to the next line, and NPE occurs